src/Microsoft.NET.Sdk.Functions.Generator/Program.cs

```diff
@@ -19,7 +24,19 @@ private static void Main(string[] args)
                 var assemblyPath = args[0].Trim();
                 var outputPath = args[1].Trim();
                 var functionsInDependencies = bool.Parse(args[2].Trim());
+                var assemblyDir = Path.GetDirectoryName(assemblyPath);
 
+#if NETCOREAPP2_1
+                AssemblyLoadContext.Default.Resolving += (context, assemblyName) =>
+                {
+                    return context.LoadFromAssemblyPath(Path.Combine(assemblyDir, assemblyName.Name + ".dll"));
+                };
+#else
+                AppDomain.CurrentDomain.AssemblyResolve += (sender, e) =>
+                {
+                    return Assembly.LoadFrom(Path.Combine(assemblyDir, e.Name + ".dll"));
+                };
```

Comment:
@fabiocav / @brettsam I'm assuming the AssemblyLoadContext.Default.Resolving is for resolving assemblies that have failed to load..? With that assumption, is this the right approach for net46?

Reply:
Oh I don't think AssemblyLoadContext works in net46. Assembly.LoadFrom() is the right approach there. By default assemblies will be searched for based on the current root... which isn't the current root of the function app. So this redirects those load requests to go to the root of the function.